cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/StreamWriterSizeCollector.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data.stream.service;
+
+/**
+ * Keep track of the sizes of the files written by one Stream handler.
+ */
+public interface StreamWriterSizeCollector {
+
+  /**
+   * Called to notify this manager that {@code dataSize} bytes of data has been ingested by the stream
+   * {@code streamName} using the stream handler from which this code is executed.
+   *
+   * @param streamName name of the stream that ingested data.
+   * @param dataSize amount of data ingested in bytes.
+   */
+  void received(String streamName, long dataSize);
```

Comment:
Can you add a note that this is incremental? The javadoc is not clear about this.